### PR TITLE
Netcore: Changes to run on linux

### PR DESCRIPTION
**Pull request by bergmania**
## Details
- Fixes path issues.
   - On Linux folder dividers `/` and not `\`. To be compatible we use `Path.Combine`
- Casing issues
   - On Linux the file system is case sensitive. So `"LazyLoad.js" != "lazyload.js` when checking for files.
   - The following npm packages, has PacalCase file names
      - Chart.js
      - LazyLoad.js
      - angular-locale_\*\*-\*\*.js (culture needs to be lowercase)
   - The `Umbraco` folder in `wwwroot`, needs to be lowercase
- Newlines
  - On windows it is `\r\n` on Linux it is `\n`.. Updated to use `Environment.NewLine`
- Serilog
  - Logging did not work before I called `UseSerilog()`, even that it works on windows.
- Lucene
  - I moved the reference to the Lucene project from the `Umbraco.Web.UI.NetCore` to the `Umbraco.Web.BackOffice`
- SqlCE 
   - Removed from NetCore executable as standard. Also updated the buildscript and SqlCE nuspec to get it from the SqlCE project instead.

# Tests:
 - Need to implement `Noop` versions of examine stuff right now
 - I registered the `Noop` versions in `AspNetCoreComposer`
```
        composition.RegisterUnique<IUmbracoIndexesCreator, NoopUmbracoIndexesCreator>();
        composition.RegisterUnique<IBackOfficeExamineSearcher, NoopBackOfficeExamineSearcher>();
```

```
    public class NoopBackOfficeExamineSearcher : IBackOfficeExamineSearcher
    {
        public IEnumerable<ISearchResult> Search(string query, UmbracoEntityTypes entityType, int pageSize, long pageIndex, out long totalFound,
            string searchFrom = null, bool ignoreUserStartNodes = false)
        {
            totalFound = 0;
            return new ISearchResult[0];
        }
    }

    public class NoopUmbracoIndexesCreator : IUmbracoIndexesCreator
    {
        public IEnumerable<IIndex> Create()
        {
            return new IIndex[0];
        }
    }
```

### CODE REVIEW

File: src/umbraco.sln [hunk 1]
```diff
@@ -59,9 +59,10 @@ Project("{E24C65DC-7377-472B-9ABA-BC803B73C61A}") = "Umbraco.Web.UI.Client", "ht
 		StartServerOnDebug = "false"
 	EndProjectSection
 EndProject
-Project("{E24C65DC-7377-472B-9ABA-BC803B73C61A}") = "Umbraco.Tests.AcceptanceTest", "Umbraco.Tests.AcceptanceTest\", "{9E4C8A12-FBE0-4673-8CE2-DF99D5D57817}"
+Project("{E24C65DC-7377-472B-9ABA-BC803B73C61A}") = "Umbraco.Tests.AcceptanceTest", "http://localhost:58896", "{9E4C8A12-FBE0-4673-8CE2-DF99D5D57817}"
```
Comment (stevetemple) [inline src/umbraco.sln:62]: Is this correct?

Comment (bergmania) [reply src/umbraco.sln:62]: I aligned with how the Umbraco.Web.UI.Client project is crated, so yes, it is intended :)

Thanks for asking. By the end of day, the important change to not having issues on linux, was the remove of these 
``` DIFF
-{9E4C8A12-FBE0-4673-8CE2-DF99D5D57817}.Debug|Any CPU.Build.0 = Debug|Any CPU
-{9E4C8A12-FBE0-4673-8CE2-DF99D5D57817}.Release|Any CPU.Build.0 = Debug|Any CPU
```

File: src/Umbraco.Tests/Logging/UmbracoTraceLog.UNITTEST.20181112.json [hunk 2]
```diff
@@ -142,7 +142,7 @@
 {"@t":"2018-11-12T08:37:57.2728260Z","@mt":"{LogPrefix} Task added {TaskType}","@l":"Debug","LogPrefix":"[KeepAlive] ","TaskType":"Umbraco.Web.Scheduling.KeepAlive","SourceContext":"Umbraco.Web.Scheduling.BackgroundTaskRunner","ProcessId":27004,"ProcessName":"iisexpress","ThreadId":9,"AppDomainId":2,"AppDomainAppId":"LMW3SVC2ROOT","MachineName":"DELLBOOK","Log4NetLevel":"DEBUG","HttpRequestNumber":11,"HttpRequestId":"47bf742b-9dd2-41f2-8e45-fc17fb44abeb"}
```
Comment (stevetemple) [inline src/Umbraco.Tests/Logging/UmbracoTraceLog.UNITTEST.20181112.json:142]: Unintentional commit?

Comment (bergmania) [reply src/Umbraco.Tests/Logging/UmbracoTraceLog.UNITTEST.20181112.json:142]: thanks

Comment (bergmania) [reply src/Umbraco.Tests/Logging/UmbracoTraceLog.UNITTEST.20181112.json]: Turned out to be required..

File: src/Umbraco.Infrastructure/Composing/CompositionExtensions/Services.cs [hunk 3]
```diff
@@ -100,9 +100,9 @@ private static LocalizedTextServiceFileSources SourcesFactory(IFactory container
         {
             var hostingEnvironment = container.GetInstance<IHostingEnvironment>();
             var globalSettings = container.GetInstance<IGlobalSettings>();
-            var mainLangFolder = new DirectoryInfo(hostingEnvironment.MapPathContentRoot(globalSettings.UmbracoPath + "/config/lang/"));
+            var mainLangFolder = new DirectoryInfo(hostingEnvironment.MapPathContentRoot(Path.Combine(globalSettings.UmbracoPath , "config","lang")));
```
Comment (Shazwazza) [inline src/Umbraco.Infrastructure/Composing/CompositionExtensions/Services.cs:103]: Although this 'works' it is incorrect. Path.Combine on windows will use backslashes which the browser and our own methods will magically fixup but we shouldn't rely on that. We cannot use Path.Combine for combining web paths. We need our own utility for combining web paths like `WebPath.Combine`, apparently you can use a trick using Uri https://stackoverflow.com/questions/372865/path-combine-for-urls

File: src/Umbraco.Infrastructure/Logging/Viewer/LogViewerConfig.cs [hunk 4]
```diff
@@ -10,7 +10,7 @@ namespace Umbraco.Core.Logging.Viewer
     public class LogViewerConfig : ILogViewerConfig
     {
         private readonly IHostingEnvironment _hostingEnvironment;
-        private const string _pathToSearches = "~/Config/logviewer.searches.config.js";
+        private static readonly string _pathToSearches = Path.Combine(Constants.SystemDirectories.Config, "logviewer.searches.config.js");
```
Comment (Shazwazza) [inline src/Umbraco.Infrastructure/Logging/Viewer/LogViewerConfig.cs:13]: Same issue previously mentioned for web paths.

File: src/Umbraco.Tests.UnitTests/Umbraco.Tests.Common/Builders/StylesheetBuilderTests.cs [hunk 5]
```diff
@@ -10,7 +11,7 @@ public class StylesheetBuilderTests
         public void Is_Built_Correctly()
         {
             // Arrange
-            const string testPath = "/css/styles.css";
+            var testPath = Path.Combine("css", "styles.css");
```
Comment (Shazwazza) [inline src/Umbraco.Tests.UnitTests/Umbraco.Tests.Common/Builders/StylesheetBuilderTests.cs:14]: Same issue previously mentioned for web paths.

File: src/Umbraco.Web.UI.NetCore/Umbraco.Web.UI.NetCore.csproj [hunk 6]
```diff
@@ -13,14 +13,25 @@
...
     <ItemGroup>
-      <ProjectReference Include="..\Umbraco.Examine.Lucene\Umbraco.Examine.Lucene.csproj" />
-      <ProjectReference Include="..\Umbraco.Persistance.SqlCe\Umbraco.Persistance.SqlCe.csproj" />
       <ProjectReference Include="..\Umbraco.PublishedCache.NuCache\Umbraco.PublishedCache.NuCache.csproj" />
       <ProjectReference Include="..\Umbraco.Web.BackOffice\Umbraco.Web.BackOffice.csproj" />
       <ProjectReference Include="..\Umbraco.Web.Common\Umbraco.Web.Common.csproj" />
       <ProjectReference Include="..\Umbraco.Web.Website\Umbraco.Web.Website.csproj" />
     </ItemGroup>
 
+    <ItemGroup>
```
Comment (Shazwazza) [inline src/Umbraco.Web.UI.NetCore/Umbraco.Web.UI.NetCore.csproj:22]: Is this all required?

File: src/Umbraco.Web.UI.NetCore/config/serilog.config [hunk 7]
```diff
@@ -19,7 +19,7 @@
         <!-- This is used by the default log viewer in the Umbraco backoffice -->
         <add key="serilog:using:File" value="Umbraco.Infrastructure" />
         <add key="serilog:write-to:File.formatter" value="Serilog.Formatting.Compact.CompactJsonFormatter, Serilog.Formatting.Compact" />
-        <add key="serilog:write-to:File.path" value="%UMBLOGDIR%\UmbracoTraceLog.%MACHINENAME%..json" />
+        <add key="serilog:write-to:File.path" value="%UMBLOGDIR%/UmbracoTraceLog.%MACHINENAME%..json" /> <!-- folder seperator need to be / and not \ to support both windows and linux-->
```
Comment (Shazwazza) [inline src/Umbraco.Web.UI.NetCore/config/serilog.config:22]: Then use a token for folder separator just like we do with the UMBLOGDIR token

File: src/Umbraco.Infrastructure/Composing/HostBuilderExtensions.cs [hunk 8]
```diff
@@ -25,6 +26,7 @@ public static IHostBuilder UseUmbraco(this IHostBuilder builder)
         /// <param name="umbracoServiceProviderFactory"></param>
         /// <returns></returns>
         public static IHostBuilder UseUmbraco(this IHostBuilder builder, UmbracoServiceProviderFactory umbracoServiceProviderFactory)
-            => builder.UseServiceProviderFactory(umbracoServiceProviderFactory);
+            => builder.UseServiceProviderFactory(umbracoServiceProviderFactory)
+                .UseSerilog();
```
Comment (Shazwazza) [inline src/Umbraco.Infrastructure/Composing/HostBuilderExtensions.cs:30]: what is this doing?

Comment (bergmania) [reply src/Umbraco.Infrastructure/Composing/HostBuilderExtensions.cs:30]: Created this follow up task: https://dev.azure.com/umbraco/D-Team%20Tracker/_workitems/edit/7757

**Comment by bergmania** [comment]
Not sure how you would create the test that ensures all paths are requested in correct casing?

File: src/Umbraco.Core/Routing/WebPath.cs [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+namespace Umbraco.Core.Routing
+{
+    public class WebPath
+    {
+        public static string Combine(params string[] paths)
+        {
+            const string separator = "/";
+
+            if (paths == null) throw new ArgumentNullException(nameof(paths));
+            if (!paths.Any()) return string.Empty;
```
Comment (Shazwazza) [inline src/Umbraco.Core/Routing/WebPath.cs:13]: should use paths.Length since it's an array instead of Linq

**Pull request merged**